spec/unit/shoulda/matchers/active_model/validate_length_of_matcher_spec.rb

```diff
@@ -370,15 +370,205 @@ def configure_validation_matcher(matcher)
+
+      it 'does not override the default message with a blank' do
+        expect(validating_length(type: :jsonb, minimum: 4)).
+          to validate_length_of(:attr).as_array.is_at_least(4).with_short_message(nil)
+      end
+
+      it 'fails when used in the negative' do
+        assertion = lambda do
+          expect(validating_length(type: :jsonb, minimum: 4)).
+            not_to validate_length_of(:attr).as_array.is_at_least(4)
+        end
+
+        message = <<-MESSAGE
+Expected Example not to validate that the length of :attr is at least 4,
+but this could not be proved.
+  After setting :attr to ‹["x", "x", "x", "x"]›, the matcher expected
+  the Example to be invalid, but it was valid instead.
+        MESSAGE
+
+        expect(&assertion).to fail_with_message(message)
+      end
+    end
+
+    context 'an attribute with a minimum length validation of 0' do
+      it 'accepts ensuring the correct minimum length' do
+        expect(validating_length(type: :jsonb, minimum: 0)).
+          to validate_length_of(:attr).as_array.is_at_least(0)
+      end
+    end
+
+    context 'an attribute with a maximum length' do
+      it 'accepts ensuring the correct maximum length' do
+        expect(validating_length(type: :jsonb, maximum: 4)).
+          to validate_length_of(:attr).as_array.is_at_most(4)
+      end
+
+      it 'rejects ensuring a lower maximum length with any message' do
+        expect(validating_length(type: :jsonb, maximum: 4)).
+          not_to validate_length_of(:attr).as_array.is_at_most(3).with_long_message(/.*/)
+      end
+
+      it 'rejects ensuring a higher maximum length with any message' do
+        expect(validating_length(type: :jsonb, maximum: 4)).
+          not_to validate_length_of(:attr).as_array.is_at_most(5).with_long_message(/.*/)
+      end
+
+      it 'does not override the default message with a blank' do
+        expect(validating_length(type: :jsonb, maximum: 4)).
+          to validate_length_of(:attr).as_array.is_at_most(4).with_long_message(nil)
+      end
+    end
+
+    context 'an attribute with a required exact length' do
+      it 'accepts ensuring the correct length' do
+        expect(validating_length(type: :jsonb, is: 4)).
+          to validate_length_of(:attr).as_array.is_equal_to(4)
+      end
+
+      it 'rejects ensuring a lower maximum length with any message' do
+        expect(validating_length(type: :jsonb, is: 4)).
+          not_to validate_length_of(:attr).as_array.is_equal_to(3).with_message(/.*/)
+      end
+
+      it 'rejects ensuring a higher maximum length with any message' do
+        expect(validating_length(type: :jsonb, is: 4)).
+          not_to validate_length_of(:attr).as_array.is_equal_to(5).with_message(/.*/)
+      end
+
+      it 'does not override the default message with a blank' do
+        expect(validating_length(type: :jsonb, is: 4)).
+          to validate_length_of(:attr).as_array.is_equal_to(4).with_message(nil)
+      end
+    end
+  end
```

Comment:
We should also add tests against an ActiveModel attribute that is marked as an array. There doesn't appear to be any tests that use an ActiveModel model in this file, so you may have to define a new method that constructs such a model instead of an ActiveRecord model. Instead of the `define_model` helper, you should be able to use the `define_active_model_class` helper to do so. See: https://github.com/jarenas9539/shoulda-matchers/blob/feature/extending-active-model-validating-length-of-to-allow-arrays/spec/unit/shoulda/matchers/active_model/validate_presence_of_matcher_spec.rb#L1134.